src/Resources/config/services.yml

```diff
@@ -35,10 +35,30 @@ services:
             # Priority must be lower than the one of the Contao initialize system listener.
             - { name: kernel.event_listener, event: kernel.request, method: onKernelRequest, priority: 28 }
 
-    contao.resource_provider:
-        class: Contao\CoreBundle\HttpKernel\Bundle\ResourceProvider
+    contao.resource_locator.bundles:
+        class: Contao\CoreBundle\Config\FileLocator
+        public: false
         arguments:
-            - "%kernel.root_dir%"
+            - "@kernel"
+        tags:
+            - { name: contao.resource_locator }
+
+    contao.resource_locator.chain:
+        class: Contao\CoreBundle\Config\ChainFileLocator
+        public: false
+
+    contao.resource_locator.cache:
+        class: Contao\CoreBundle\Config\CombinedFileLocator
+        public: false
+        arguments:
+            - '%kernel.cache_dir%/contao'
+            - "@contao.resource_locator"
+
+    contao.resource_locator:
```

Comment:
The alias is what should actually be used in the code. The others are pre-definitions for the container builder pass. Depending on if there are loaders or not, the alias will be changed. This is the same as Twig does with it's loaders.